src/launchpad/sentry_client.py

```diff
@@ -138,53 +142,65 @@ def __init__(self, base_url: str, shared_secret: str | None = None) -> None:
         url = self._build_url(endpoint)
+
+        total_size = None
+        try:
+            head_resp = self.session.head(url, auth=self.auth, timeout=30)
+            if head_resp.status_code == 200:
+                content_length = head_resp.headers.get("Content-Length")
+                if content_length:
+                    total_size = int(content_length)
+        except Exception:
+            pass
+
         file_size = 0
+        chunk_size = 20 * 1024 * 1024  # 20MB chunks
+        chunk_count = 0
 
-        attempts = 3
-        for attempt in range(attempts):
+        for attempt in range(RETRY_ATTEMPTS):
             try:
-                response = self.session.get(url, auth=self.auth, timeout=120, stream=True)
-                if response.status_code != 200:
+                headers = {}
+                if file_size > 0:
+                    headers = {"Range": f"bytes={file_size}-"}
+
+                response = self.session.get(url, auth=self.auth, headers=headers, timeout=120, stream=True)
+
+                if response.status_code not in (200, 206):
                     raise SentryClientError(response=response)
-                for chunk in response.iter_content(chunk_size=8192):
+
+                for chunk in response.iter_content(chunk_size=chunk_size):
                     if chunk:
                         out.write(chunk)
                         file_size += len(chunk)
-                        if file_size > 5 * 1024 * 1024 * 1024:
-                            raise SentryClientError(
-                                detail="Failed to download artifact: File size exceeds 5GB limit",
-                                response=response,
+                        chunk_count += 1
+
+                        if total_size:
+                            total_chunks = (total_size + chunk_size - 1) // chunk_size
+                            progress = (file_size / total_size) * 100
+                            logger.info(
+                                f"Downloaded chunk {chunk_count}/{total_chunks} - {file_size / MB_DIVISOR:.1f} MB ({progress:.1f}%)"
                             )
-                break
+                        else:
+                            logger.info(f"Downloaded chunk {chunk_count} - {file_size / MB_DIVISOR:.1f} MB")
+
+                return file_size
+
             except (ConnectionError, Timeout, ChunkedEncodingError, ContentDecodingError) as e:
-                if attempt == 0:
-                    logger.warning(f"Download failed due to network error ({attempt + 1}/{attempts})", exc_info=True)
-                else:
-                    logger.exception(f"Download failed after {attempts} attempts due to network error")
-                    raise SentryClientError(
-                        detail="Failed to download artifact (network error)", response=response, exception=e
+                if attempt < RETRY_ATTEMPTS - 1:
+                    wait_time = 2**attempt  # Exponential backoff
```

Comment:
N is so small this won't really provide exponential backoff like you'd expect. It might be better to just use a fixed wait time like 5 seconds.